Amazon.QLDB.Driver/driver/AsyncQldbDriver.cs

```diff
@@ -52,15 +72,24 @@ public static AsyncQldbDriverBuilder Builder()
         /// </summary>
         public void Dispose()
         {
-            this.sessionPool.Dispose();
+            if (!this.isClosed)
+            {
+                this.isClosed = true;
+                while (this.sessionPool.Count > 0)
+                {
+                    this.sessionPool.Take().Close();
```

Comment:
Are you sure this will not block?

Reply:
Yup. It throws an exception if it is empty, and typically in .NET it's idiomatic to not call Dispose more than once.